internal/proposals/migration-hooks/migration-hooks-proposal.md

```diff
@@ -35,7 +35,7 @@ To fulfill the requirements, a new package, called `JobManager`, is introduced,
 
 Furthermore, the `JobManager` package has a `duration` variable for benchmarking.
 
-Jobs are implemented within the `JobManager` package in `go`-files, one for each component, using the specific `job`-interface. Then, the implemented interface is registered using `register(job)` in the same file.
+Jobs are implemented within the `JobManager` package in `go`-files, one for each component, using the specific `job`-interface. Then, the implemented interface is registered using `register(job)` in the same file. To implement the `job`-interface, the newly created jobs need to implement the `execute(*config.Config, kubernetes.Interface)` function, which takes the installation config and an kubernetes interface as input, so that the jobs can interact with the cluster. The return value needs to be an error. Additionally the `when()` function needs to be implemented, which returns the component the job is bound to and if it should run pre or post the deployment. And if the active solution for tagging jobs as deprecated is choosen, then the `deprecate` function also needs to be implemented - more in the next section.
```

Comment:
```suggestion
Jobs are implemented within the `JobManager` package in `go`-files, one for each component, using the specific `job` interface. Then, the implemented interface is registered using `register(job)` in the same file.
To implement the `job` interface, the newly created jobs must implement the `execute(*config.Config, kubernetes.Interface)` function, which takes the installation config and a kubernetes interface as input, so that the jobs can interact with the cluster. The return value must be an error. Additionally, the `when()` function must be implemented, which returns the component the job is bound to and whether it should run pre or post the deployment. If the active solution for tagging jobs as deprecated is chosen, then the `deprecate` function also must be implemented - more in the next section.
```